### PR TITLE
Refactor from NODE_ENV to OC_ENV

### PR DESCRIPTION
**Pull request by znarf**
following directions from https://github.com/vercel/next.js/blob/master/errors/non-standard-node-env.md

- `NODE_ENV` should only be production/development/test
- `OC_ENV` should be production/staging/e2e/ci/test/development

What's changing:
- staging: `NODE_ENV=production OC_ENV=staging` (before `NODE_ENV=production OC_ENV=production`)
- e2e: `NODE_ENV=production OC_ENV=e2e` (before `NODE_ENV=e2e OC_ENV=e2e`)
- ci: `NODE_ENV=production OC_ENV=ci` (before `NODE_ENV=ci OC_ENV=ci`)

What's not changing:
- production: `NODE_ENV=production OC_ENV=production`
- test: `NODE_ENV=test OC_ENV=test`
- development: `NODE_ENV=development OC_ENV=development`

Todo:
- configure `OC_ENV` in production (default to NODE_ENV anyway)
- configure `OC_ENV` in staging

### CODE REVIEW

**Review by Betree — APPROVED** [review]
Great change, it's also going to provide better ways to detect the env for https://github.com/opencollective/opencollective-frontend/blob/ae407e54fefe7a131877b7e08541e138da1c51c6/server/content-security-policy.js#L106 and https://github.com/opencollective/opencollective-frontend/blob/587a16813442a5a0172bb7801b05931f9127f12e/server/sentry.js#L10.

Would be great to have the same behavior for API.

File: server/content-security-policy.js [hunk 1]
```diff
@@ -102,44 +102,35 @@ const getContentSecurityPolicyConfig = () => {
...
+  } else {
+    // Third party deploy, or Zeit deploy preview
+    return {
+      reportOnly: true,
+      directives: generateDirectives(),
+    };
   }
-
-  // Disabled in other environments
-  return false;
```
Comment (znarf) [inline server/content-security-policy.js:142]: @Betree That is removed in my version, not sure if you saw it. Let me know what you think is best for `test`, `e2e` and `ci`? And what should be the default?

Comment (Betree) [reply server/content-security-policy.js:142]: I missed that, perfect!

Comment (Betree) [reply server/content-security-policy.js:142]: The default should probably be `development` I guess

**Pull request merged**